code/WorkInProgress/TempEngine.dm

```diff
@@ -143,13 +156,23 @@
 		window = new(src)
 
 		SPAWN_DBG(0.5 SECONDS)
-			circ1 = locate(/obj/machinery/atmospherics/binary/circulatorTemp) in get_step(src,WEST)
-			circ2 = locate(/obj/machinery/atmospherics/binary/circulatorTemp) in get_step(src,EAST)
-			if(!circ1 || !circ2)
-				status |= BROKEN
+			src.circ1 = locate(/obj/machinery/atmospherics/binary/circulatorTemp) in get_step(src,WEST)
+			src.circ2 = locate(/obj/machinery/atmospherics/binary/circulatorTemp) in get_step(src,EAST)
+			if(!src.circ1 || !src.circ2)
+				src.status |= BROKEN
+
```

Comment:
I feel like it would make a lot of sense to set `circ1.side` and `circ2.side` here. (For example the current setup possibly breaks on pamgoC.)
Alternative solution is to get rid of `var/side` altogether and just check `if(src.generator.circ1 == src)` etc. in circulator's disposing.

Reply:
side is used in the iconstates which is kind of nice since the right sprites will get used now on pamgoC

side is now null until set by generator.